### ISSUE TITLE
Angular global config for providedIn and namedClient

### ISSUE THREAD

**Issue by fridolin-koch** [dotansimha/graphql-code-generator#1469]
**Is your feature request related to a problem? Please describe.**

I'm using nrwl/nx to encapsulate different API clients into libraries. This leads to every query and mutation being annotated with two directives.

**Describe the solution you'd like**

It would be easier and way cleaner to provide these options globally inside the configuration file, e.g.:

```yaml
overwrite: true
schema: http://localhost:3000/cms/query
documents: lib/graphql/*.graphql
generates:
  ./lib/generated/graphql.ts:
    config:
      nameClient: 'my-lib'
      ngModule: './my-client.module#MyClientModule"
    plugins:
      - add: '/* tslint:disable */'
      - typescript-common
      - typescript-client
      - typescript-apollo-angular
```

**Describe alternatives you've considered**

Can't think of any clean alternative.

**Additional context**

I'm happy to provide an PR if the solution would be feasible.

Best regards,
Frido

P.S.: Awesome project :)

**Comment by dotansimha** [comment]
@kamilkisiela

**Comment by kamilkisiela** [comment]
https://github.com/dotansimha/graphql-code-generator/pull/1477

**Comment by dotansimha** [comment]
Available in 1.0.0 👍
